pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1743,3 +1747,13 @@ func (f *schemaChangeFrontier) schemaChangeBoundaryReached() (r bool) {
 		f.latestTs.Equal(f.boundaryTime) &&
 		f.boundaryType != jobspb.ResolvedSpan_NONE
 }
+
+// laggingHighwater returns true when the time between the earliest and latest
+// resolved spans has exceeded the configured HighwaterLagCheckpointThreshold
+func (f *schemaChangeFrontier) laggingHighwater(sv *settings.Values) bool {
```

Comment:
nit: not sure I like the name.  kinda implies that we should return  Time, but we return bool.
how about `haveLaggingSpans()` or something similar?